core/lib/dal/src/tee_proof_generation_dal.rs

```diff
@@ -64,72 +64,93 @@ impl TeeProofGenerationDal<'_, '_> {
     ) -> DalResult<Option<LockedBatch>> {
         let processing_timeout = pg_interval_from_duration(processing_timeout);
         let min_batch_number = i64::from(min_batch_number.0);
+        let mut transaction = self.storage.start_transaction().await?;
+        let batch_number = sqlx::query_as!(
+            StorageBatch,
+            r#"
+            SELECT
+                p.l1_batch_number
+            FROM
+                proof_generation_details p
+            LEFT JOIN
+                tee_proof_generation_details tee
+                ON
+                    p.l1_batch_number = tee.l1_batch_number
+                    AND tee.tee_type = $1
+            WHERE
+                (
+                    p.l1_batch_number >= $5
+                    AND p.vm_run_data_blob_url IS NOT NULL
+                    AND p.proof_gen_data_blob_url IS NOT NULL
+                )
+                AND (
+                    tee.l1_batch_number IS NULL
+                    OR (
+                        (tee.status = $2 OR tee.status = $3)
+                        AND tee.prover_taken_at < NOW() - $4::INTERVAL
+                    )
+                )
+            FETCH FIRST ROW ONLY
+            "#,
+            tee_type.to_string(),
+            TeeProofGenerationJobStatus::PickedByProver.to_string(),
+            TeeProofGenerationJobStatus::Failed.to_string(),
+            processing_timeout,
+            min_batch_number
+        )
+        .instrument("lock_batch_for_proving#get_batch_no")
+        .with_arg("tee_type", &tee_type)
+        .with_arg("processing_timeout", &processing_timeout)
+        .with_arg("min_batch_number", &min_batch_number)
+        .fetch_optional(&mut transaction)
+        .await?;
+
+        let batch_number = match batch_number {
+            Some(batch) => batch.l1_batch_number,
+            None => {
+                transaction.commit().await?;
```

Comment:
There are seemingly no changes in the transaction at this point. Is the commit here just to not clean up DB resources faster?

Reply:
To be honest, I thought every `.start_transaction()` needed to either end with `.commit()` or `.rollback()`. I'm not sure how you determine that it's OK to skip it altogether, but I trust your judgment here. I this line [here](https://github.com/matter-labs/zksync-era/pull/3342/commits/464b5d4f1a189304fc442eebe847fee2a5d714d5#diff-46d00b49fb68d6da13c9ecdc5ab3dab6adf502ec216349721586e96553652444L111). ✅

Reply:
A transaction that is dropped w/o being committed is rolled back implicitly. You can add an explicit call to be more precise (but IMO it should be `rollback` so that it's clear that no modifying actions have been performed).

Reply:
I'm fine with rolling back the transaction implicitly then. 👍🏼